packages/node/src/transports/setup.ts

```diff
@@ -0,0 +1,52 @@
+import { getEnvelopeEndpointWithUrlEncodedAuth, initAPIDetails, NewTransport, NoopTransport } from '@sentry/core';
+import { Transport, TransportOptions } from '@sentry/types';
+import { makeDsn } from '@sentry/utils';
+
+import { NodeOptions } from '../types';
+import { HTTPSTransport, HTTPTransport, makeNodeTransport } from '.';
+
+/**
+ * Sets up Node transport based on the passed `options`.
+ *
+ * @returns an object currently still containing both, the old `Transport` and
+ * `NewTransport` which will eventually replace `Transport`. Once this is replaced,
+ * this function will return a ready to use `NewTransport`.
+ */
+// TODO(v7): Adjust return value when NewTransport is the default
+export function setupNodeTransport(options: NodeOptions): { transport: Transport; newTransport?: NewTransport } {
```

Comment:
Yes these `setupXTransport` functions will get much smaller, once the old transports are removed, agreed. Let's see how/where we handle ClientOptions. This will probably be the place to set up transports then (or pass on a custom one).

Reply:
And regarding setup tests: They don't have to stay. I just quickly added them to test setup functionality explicitly. Kinda makes sense at least for fetch vs xhr on the browser side IMO. But let's see if/where they stay when we declutter/restructure the transport setup